models/issue_label.go

```diff
@@ -325,6 +330,129 @@ func GetLabelsByRepoID(repoID int64, sortType string, listOptions ListOptions) (
 	return getLabelsByRepoID(x, repoID, sortType, listOptions)
 }
 
+// ________
+// \_____  \_______  ____
+//  /   |   \_  __ \/ ___\
+// /    |    \  | \/ /_/  >
+// \_______  /__|  \___  /
+//         \/     /_____/
+
+// getLabelInOrgByName returns a label by Name in given organization
+// If pass orgID as 0, then ORM will ignore limitation of organization
+// and can return arbitrary label with any valid ID.
+func getLabelInOrgByName(e Engine, orgID int64, labelName string) (*Label, error) {
+	if len(labelName) == 0 {
+		return nil, ErrOrgLabelNotExist{0, orgID} // FIX ME!
+	}
+
+	l := &Label{
+		Name:  labelName,
+		OrgID: orgID,
+	}
+	has, err := e.Get(l)
+	if err != nil {
+		return nil, err
+	} else if !has {
+		return nil, ErrOrgLabelNotExist{0, l.OrgID}
+	}
+	return l, nil
+}
+
+// getLabelInOrgByID returns a label by ID in given organization.
+// If pass orgID as 0, then ORM will ignore limitation of organization
+// and can return arbitrary label with any valid ID.
+func getLabelInOrgByID(e Engine, orgID, labelID int64) (*Label, error) {
+	if labelID <= 0 {
+		return nil, ErrOrgLabelNotExist{labelID, orgID}
+	}
+
+	l := &Label{
+		ID:    labelID,
+		OrgID: orgID,
+	}
+	has, err := e.Get(l)
```

Comment:
Shouldn't we better check that `l.repoID == 0`?

Reply:
Sure, I've added RepoID: 0 to the label we create above that is passed to db search (and added separate check for org id >= 0 already per previous suggestion).